### DISCUSSION TITLE
Add Google Health Connect Support, Get blood sugar data out of google health connect, Add blood sugar data as data_type for watchface widgets

### DISCUSSION THREAD

**Discussion by 4RK4N** [zepp-health/zeppos-forum#84] [Developer Feedback]
well hi,
the title basically says it all.
there is new "google health connect" https://play.google.com/store/apps/details?id=com.google.android.apps.healthdata&hl=de&gl=US which is an on-device storage for all sorts of health data. my current blood sugar recieving setup can already push the blood sugar data into google health connect.
now what i want as a feature request: 
add support for google health connect, retrieve the blood sugar data from there, display it in the app and add it as data_type for watchface widgets.
that way many diabetics could be helped by having a reliable way to check their values on the go.

**Comment by MyoungXUE** [comment]
Hello, I will feedback your suggestion to the relevant colleagues.

**Reply by oakeley** [reply]
This would be a big improvement, really support this initiative. Thanks

**Comment by raylewisdesign** [comment]
We appreciate your suggestion. Unfortunately, Zepp does not have a plan to connect with Google Health Services yet. As an alternative, You can use the Zepp app or download a blood sugar recording app on your watch to track your blood sugar data.

**Reply by oakeley** [reply]
@raylewisdesign not sure that you have understood the request

**Reply by 4RK4N** [reply]
just fyi the blood sugar is read with a medical sensor every 5 minutes and pushed into google health connect already. 
so manually entering a bg value either on the watch or the app every 5 minutes is not a viable option.

**Reply by raylewisdesign** [reply]
In fact, there are three issues:
Firstly, syncing Google services within Zepp's ecosystem is already a complex problem. 
Secondly, maintaining continuous background running of the Zepp app and transmitting data to the watch is also challenging. 
Thirdly, since it is necessary to ensure high-frequency data exchange, a better solution is to connect directly with the Glucose meter via a JS app on the watch.
What do you think?

**Comment by dlgair** [comment]
> well hi, the title basically says it all. there is new "google health connect" https://play.google.com/store/apps/details?id=com.google.android.apps.healthdata&hl=de&gl=US which is an on-device storage for all sorts of health data. my current blood sugar recieving setup can already push the blood sugar data into google health connect. now what i want as a feature request: add support for google health connect, retrieve the blood sugar data from there, display it in the app and add it as data_type for watchface widgets. that way many diabetics could be helped by having a reliable way to check their values on the go.

This is a fabulous idea. :-)
